ibm/data_source_ibm_iam_roles.go

```diff
@@ -0,0 +1,100 @@
+package ibm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/IBM-Cloud/bluemix-go/api/iampap/iampapv2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func datasourceIBMIAMRole() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceIBMIAMRoleRead,
+
+		Schema: map[string]*schema.Schema{
+			"service": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The Service Name",
+				ForceNew:    true,
+			},
+			"roles": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+
+}
+
+func datasourceIBMIAMRoleRead(d *schema.ResourceData, meta interface{}) error {
+	iampapv2Client, err := meta.(ClientSession).IAMPAPAPIV2()
+	if err != nil {
+		return err
+	}
+
+	userDetails, err := meta.(ClientSession).BluemixUserDetails()
+	if err != nil {
+		return err
+	}
+
+	var serviceName string
+	var customRoles, serviceRoles, systemRoles []iampapv2.Role
+	if service, ok := d.GetOk("service"); ok {
+		serviceName = service.(string)
+
+		customRoles, err = iampapv2Client.IAMRoles().ListCustomRoles(userDetails.userAccount, serviceName)
+		if err != nil && !strings.Contains(err.Error(), "404") {
+			return fmt.Errorf("Error retrieving Custom Roles: %s", err)
```

Comment:
just return the error no need to check the condition !strings.Contains(err.Error(), "404"